### ISSUE TITLE
[v0.2.0] Release Tracker 

### ISSUE THREAD

**Issue by guangy10** [pytorch/executorch#2666]
## The [v0.2.0](https://github.com/pytorch/executorch/releases/tag/v0.2.0) release has been cut, and this issue is closed. Use https://github.com/pytorch/executorch/issues/3409 for cherry-picking into the v0.2.1 release.

-----

v0.2.0 release branch was cut. The branch name is "[release/0.2](https://github.com/pytorch/executorch/tree/release/0.2)", with [tag "v0.2.0-rc4"](https://github.com/pytorch/executorch/tree/v0.2.0-rc4) atm. 

Our release plan from this point is roughly:

* **Phase 1 (until 4/5/24):**
-- work on finalizing the release branch for stability for *soft-launch*.
-- initial set of documentations and reproducible instructions, including Llama.
-- apply release-branch specific changes (changes not in `main` but directly in `release/0.2` branch like pins).
-- Coordinate with [PyTorch v2.3 release milestones](https://github.com/pytorch/pytorch/issues/121760), and prioritize to get PyTorch PRs needed by ExecuTorch v0.2.0 cherry-picked into PyTorch release v2.3.0. It will be very hard to do so after 04/01.

* **Phase 2 (after 4/5/24, until 4/24/24 by 12:00pm PDT):** 
--  perform extended integration/stability/performance testing based on Release Candidate builds.
--  fix critical bugs based on early feedback from our *soft-launch*
--  enhance our documentation.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

Focus on stability. Ideally only low risk changes may be cherry-picked from main:

1. Critical fixes in core components: [build](https://github.com/pytorch/executorch/tree/main/build), [exir](https://github.com/pytorch/executorch/tree/main/exir), [backends](https://github.com/pytorch/executorch/tree/main/backends), [runtime](https://github.com/pytorch/executorch/tree/main/runtime), [sdk](https://github.com/pytorch/executorch/tree/main/sdk)
2. Bug fixes in demos/examples. No new features/experiments
3. Critical bug fixes based on community feedback
4. Documentation improvements
5. Test/CI fixes
6. Release branch specific changes (e.g. change version identifiers)

Any core components changes and other change will require special dispensation from the release managers (currently  @guangy10 @mergennachin @dbort ). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the release branch. 
   **Option 1**: Use `pytorchbot` to cherry pick a PR that has been committed to the main branch using `@pytorchbot cherry-pick` command as follows.
     
      ```
      @pytorchbot cherry-pick --onto release/0.2 -c {regression,critical,fixnewfeature,docs,release}
      ```
     
     <details>
     
      ```
      usage: @pytorchbot cherry-pick --onto ONTO [--fixes FIXES] -c
                               {regression,critical,fixnewfeature,docs,release}

      Cherry pick a pull request onto a release branch for inclusion in a release

      optional arguments:
        --onto ONTO           Branch you would like to cherry pick onto (Example: release/0.2)
        --fixes FIXES         Link to the issue that your PR fixes (i.e. https://github.com/pytorch/executorch/issues/2345)
        -c {regression,critical,fixnewfeature,docs,release}
                        A machine-friendly classification of the cherry-pick reason.
                        
      ```
      
      </details>
      
      **Option 2**: Manually create a cherry pick PR using git command as follows.
      <details>

      ```
      # Find the hash of the commit you want to cherry pick
      # (for example, abcdef12345)
      git log

      git fetch origin release/0.2
      git checkout release/0.2
      git cherry-pick -x abcdef12345

      # Submit a PR based against 'release/0.2' either:
      # via the GitHub UI
      git push my-fork

      # via the GitHub CLI
      gh pr create --base release/0.2
      
      ```
      
      </details>

3. Make a request below with the following format:

```
Link to landed trunk PR:
* 

Link to release branch PR:
* 

Criteria category and reasons:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud.pytorch.org/hud/pytorch/executorch/release%2F0.2)

### Versions

0.2.0

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2611

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2726

Criteria category and reasons:
* Critical. This is used to inspect graph breaks and delegation summary, and we need this for the User Journey.

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/2696

Lik to release branch PR:

https://github.com/pytorch/executorch/pull/2746

Criteria category and reasons:

- Critical. Resolves missing error messages on build when missing dependencies and building.

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

https://github.com/pytorch/executorch/pull/2687
Lik to release branch PR:

https://github.com/pytorch/executorch/pull/2745
Criteria category and reasons:

Critical. Resolves build failures on mac build under conda when not explicitly specifying PYTHON_EXECUTABLE. See blocker task [T183430503](https://www.internalfb.com/intern/tasks/?t=183430503).

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/2723

Link to release branch PR:

- https://github.com/pytorch/executorch/pull/2749 

Criteria category and reasons:

- Docs: Restore build instructions in CMakeLists.txt top-level comment.

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2919

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2923

Criteria category and reasons:
* Documentation improvements. Better readability for markdown.

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2899

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2924

Criteria category and reasons:
* Documentation improvements.  For Android, only use path `/data/local/tmp/llama` to avoid confusion.

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/2911

Link to release branch PR:

- https://github.com/pytorch/executorch/pull/2939

Criteria category and reasons:

- Documentation improvements and critical fixes for LLM getting started guide.

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2927

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2929

Criteria category and reasons:
* Update documentation for Llama2

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2926

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2942

Criteria category and reasons:
* Update documentation fix for Llama2

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2921

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2942

Criteria category and reasons:
* Update documentation for Llama2

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2932

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2943

Criteria category and reasons:
* Fix speed generation calculation for iLlama app

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/2570

Link to release branch PR:
https://github.com/pytorch/executorch/pull/2945

Criteria category and reasons:
Fix constant tagging, needed by qnn/coreml/mps partitioner

**Comment by cccclai** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2876

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2946

Criteria category and reasons:
Exclude mutable buffers in constant tagging

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/2911

Link to release branch PR:

- https://github.com/pytorch/executorch/pull/2939

Criteria category and reasons:

- Documentation updates to LLM getting started guide to fix blockers in ExecuTorch installation on mac when not using Conda. Minor other doc fixes.

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/2940

Link to release branch PR:

- https://github.com/pytorch/executorch/pull/2959

Criteria category and reasons:

- Documentation updates to LLM getting started guide to fix blockers in ExecuTorch installation on mac when not using Conda. Minor other doc fixes.

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/2868

Link to release branch PR:

- https://github.com/pytorch/executorch/pull/2960

Criteria category and reasons:

- Support auto-downloading buck from CMake on Intel Macs. Needed to streamline CMake builds in OSS, since without this, users will have to download the appropriate version of buck2 and explicitly pass in the path.

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2952

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2971

Criteria category and reasons:
* Doc changes, refine The LLM Manual

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2992

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2993

Criteria category and reasons:
* Doc improvement

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2934
* https://github.com/pytorch/executorch/pull/2961
* https://github.com/pytorch/executorch/pull/2962
* https://github.com/pytorch/executorch/pull/2956

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2995

Criteria category and reasons:
* Bug fixes in demos/examples.

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2977

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/2997

Criteria category and reasons:
* Add a missing import in the LLM manual doc

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/2957

Link to release branch PR:

https://github.com/pytorch/executorch/pull/2999

Criteria category and reasons:

Fix qnn quantizer when it tries to quantize non fp32 tensor

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2980

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3020

Criteria category and reasons:
* Test/CI fix

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2956

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3026

Criteria category and reasons:
* Bug fixes in demos/examples. No new features/experiments. Need #3025 to be merged first. Self contained.

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2962

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3027

Criteria category and reasons:
* Bug fixes in demos/examples. No new features/experiments. Need #3025 and #3026 to be merged first. Self contained.

**Comment by Gasoonjia** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/2973

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3042

Criteria category and reasons:

Doc changes, refine The LLM Manual

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3005

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3052

Criteria category and reasons:

Getting ready to push the changes to enable running stories end to end for qnn backend. Other prs includes https://github.com/pytorch/executorch/pull/3007, https://github.com/pytorch/executorch/pull/3036, https://github.com/pytorch/executorch/pull/3037, https://github.com/pytorch/executorch/pull/3038. Will submit merged PRs accordingly when this one merges

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3007

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3061

Criteria category and reasons:

Getting ready to push the changes to enable running stories end to end for qnn backend. https://github.com/pytorch/executorch/pull/3005 is merged, this is the second pr. Remaining ones are https://github.com/pytorch/executorch/pull/3036, https://github.com/pytorch/executorch/pull/3037, https://github.com/pytorch/executorch/pull/3038 (the last pr for qnn). Will submit merged PRs accordingly when this one merges

**Comment by svekars** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3045

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3098

Criteria category and reasons:

Doc build fix. This one needs to be cherry-picked before #3099.

**Comment by svekars** [comment]
Link to landed trunk PR:

https://github.com/pytorch/executorch/pull/3071

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3099

Criteria category and reasons:

Doc build fix. This one needs to be cherry-picked after  https://github.com/pytorch/executorch/pull/3098.

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3036

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3114

Criteria category and reasons:

Remaining changes to enable qnn are https://github.com/pytorch/executorch/pull/3037, https://github.com/pytorch/executorch/pull/3038. Will submit merged PRs accordingly when this one merges

**Comment by SS-JIA** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3113

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3124

Criteria category and reasons:
* Documentation; this PR adds documentation and tutorials for the ExecuTorch Vulkan backend.

**Comment by cymbalrush** [comment]
Link to landed trunk PR:
- https://github.com/pytorch/executorch/pull/2984

Link to release branch PR:
- https://github.com/pytorch/executorch/pull/3134

Criteria category and reasons:
- The PR has fixes for running the llama model using the Core ML backend.

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3037

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3166

Criteria category and reasons:

Remaining changes to enable qnn are https://github.com/pytorch/executorch/pull/3038. Will submit merged PRs accordingly when this one merges

**Comment by Gasoonjia** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3152
Link to release branch PR:

https://github.com/pytorch/executorch/pull/3175
Criteria category and reasons:
Unblock nanogpt for dynamic export

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3146

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3178

Criteria category and reasons:

Docs for lowering smaller models to mps/coreml/qnn

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3038

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3182

Criteria category and reasons:

last pr to have qnn working end to end

**Comment by larryliu0820** [comment]
Link to landed trunk PR:
* #2996 

Link to release branch PR:
* #3186

Criteria category and reasons:
* Fix critical feature https://github.com/pytorch/executorch/actions/runs/8765991204/job/24057585488

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3039

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3191

Criteria category and reasons:
* This fixes SDK which is broken at ETRecord graph deserialization

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/2998

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3192

Criteria category and reasons:
* This fixes a bug - mishandles empty tensors - of the SDK

**Comment by cccclai** [comment]
Fix the build-frameworks-ios CI failure in release/0.2

https://github.com/pytorch/executorch/pull/3190

just so https://github.com/pytorch/executorch/pull/3134 can be merged

**Comment by dbort** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3137

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3218

Criteria category and reasons:
* Fixes to make pybindings work when installed via pip

**Comment by dbort** [comment]
Link to landed trunk PR:
* None; I will cherry-pick these back to main later

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3049
  * This PR has some failing jobs, but it's a know issue. ExecuTorch doesn't work for Python 3.8 or 3.12, but there is no way to disable those jobs yet. I am working with the devinfra team to add a way to skip these jobs in the future.

Criteria category and reasons:
* Fixes dynamic linking issues in the pip wheels built by CI jobs (T185860350)

**Comment by GregoryComer** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/3216

Link to release branch PR:

- https://github.com/pytorch/executorch/pull/3226

Criteria category and reasons:

- Docs: Remove unused README which is keeping aot_util directory alive. Debatable whether this should be cherry-picked onto release, but the README in aot_util does contain out of date information.

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3212

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3230

Criteria category and reasons:
* Docs: update outdated info on SDK docs

**Comment by svekars** [comment]
Link to landed trunk PR:

https://github.com/pytorch/executorch/pull/3153

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3243

Criteria category and reasons:

Docs: fix to the doc build to enable tags landing instead of branches. Also, corrected the logic to display version in the dropdown.

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3222

Link to release branch PR:

https://github.com/pytorch/executorch/pull/3248

Criteria category and reasons:

Fix: Fix the mps delegate runner and the pybind with mps

**Comment by cymbalrush** [comment]
Link to landed trunk PR:

- https://github.com/pytorch/executorch/pull/3188
Link to release branch PR:

- https://github.com/pytorch/executorch/pull/3249

Criteria category and reasons:

- Updated Core ML backend docs.

**Comment by Gasoonjia** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3224
Link to release branch PR:

https://github.com/pytorch/executorch/pull/3252
Criteria category and reasons:

doc update

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3246

Link to release branch PR:
https://github.com/pytorch/executorch/pull/3258

Criteria category and reasons:

update sdk delegate doc

**Comment by dbort** [comment]
Link to landed trunk PR:
* None; I will cherry-pick these back to main later

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3259

Criteria category and reasons:
* Dynamically set the pip wheel version from the repo, branch, and build environment

**Comment by mcr229** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3237?fbclid=IwAR1WQgPtuyb_D2ByuW9pbNXdoNBAeq99Uim-PTiH-DzNMJ-6Ucm7AT_T8QY

Link to release branch PR: 
https://github.com/pytorch/executorch/pull/3261

Criteria Category and reasons:
Update XNNPACK Docs and Readme

**Comment by kirklandsign** [comment]
Release branch only PR
To build artifacts for android prebuilt library 
https://github.com/pytorch/executorch/pull/3255

**Comment by manuelcandales** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3100

Link to release branch PR:
https://github.com/pytorch/executorch/pull/3253

Criteria category and reasons:
Fixing bug on sub op, introduced by a recent refactor

**Comment by cccclai** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3260

Link to release branch PR:
https://github.com/pytorch/executorch/pull/3271

Criteria category and reasons:
Fix the link in the code

**Comment by dbort** [comment]
Link to landed trunk PR:
* None; I will cherry-pick this back to main later

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3265

Criteria category and reasons:
* Clean up the pip package metadata in preparation for pushing to PyPI

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3228

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3283

Criteria category and reasons:
* Bug fixes in demos/examples

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* None. This doesn't apply to main branch. Main branch doesn't have this issue as ops are registered in a different way.

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3284

Criteria category and reasons:
* Bug fixes in demos/examples

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3285

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3286

Criteria category and reasons:
* Docs update

**Comment by kirklandsign** [comment]
Link to landed trunk PR:
* None. This doesn't apply to main branch. Main branch doesn't have this issue as ops are registered in a different way.

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3284 [NOTE: Please merge #3288 first - this is on top of it]

Criteria category and reasons:
* Bug fixes in demos/examples

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3238

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3292

Criteria category and reasons:
* SDK doc change

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3250

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3293

Criteria category and reasons:
* Fix doc

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3254

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3294

Criteria category and reasons:
* Doc change

**Comment by tarun292** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3240

Link to release branch PR:
https://github.com/pytorch/executorch/pull/3297

Criteria category and reasons:

Currently in the CoreML delegate SDK integration the time scale for the timestamps that show up for the delegated events are off and not consistent with the timescales of the CPU events. Added a callback to Inspector to convert this into the right time value. This should be a low risk change as existing functionality isn't effected by this but only users who leverage this callback will be effected. Was also validated by @cymbalrush.

**Comment by Gasoonjia** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3176
Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3307

Criteria category and reasons:
* merge nanogpt code into executorch example.

**Comment by Gasoonjia** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3202

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3308

Criteria category and reasons:
* Update llm manual doc to support dynamic export

**Comment by dbort** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3244

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3310

Criteria category and reasons:
* Fixes docs links in the LLM manual

**Comment by dbort** [comment]
Link to landed trunk PR:
* None; I will cherry-pick some of this back to main later

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3277

Criteria category and reasons:
* Make the executorch pip package depend on torch==2.3

**Comment by Olivia-liu** [comment]
Request for cherry-picking into **release/0.2**. This is a weird one:

PR https://github.com/pytorch/executorch/pull/3229 was created from exporting D56447799. D56447799 is landed in fbcode, and the commit, which has 2 files changes, was landed in main branch in OSS as well, but the PR is still open. I don't care what we do with the PR, but I want the 2 files changes to be merge it into **release/0.2**, any help?

**Comment by lucylq** [comment]
Link to landed trunk PR:

https://github.com/pytorch/executorch/pull/3315
Link to release branch PR:

[placeholder]
Criteria category and reasons:

docs

**Comment by Gasoonjia** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3300

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3321

Criteria category and reasons:
* docs

**Comment by shoumikhin** [comment]
https://github.com/pytorch/executorch/pull/3267 introduced a regression with MPS export, the team is on it and should provide a hotfix soon.

**Comment by larryliu0820** [comment]
Link to landed trunk PR:

https://github.com/pytorch/executorch/pull/3320
Link to release branch PR:

[placeholder]
Criteria category and reasons:

docs

**Comment by shoumikhin** [comment]
Link to landed trunk PR:
https://github.com/pytorch/executorch/pull/3322

Link to release branch PR:
[placeholder]

Criteria category and reasons:
Docs

**Comment by Gasoonjia** [comment]
Link to landed trunk PR:
[placeholder]
Link to release branch PR:

[placeholder]
Criteria category and reasons:

bring buck2 installation back to unblock user running buck2 examples

**Comment by dbort** [comment]
Link to landed trunk PR:
* None; I will cherry-pick this back to main later

Link to release branch PR:
* [placeholder]

Criteria category and reasons:
* Update README-wheel.md to document pybindings restrictions (which backends and operators are linked in)

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* N/A

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3306

Criteria category and reasons:
* Documentation to reflect checking out latest stable branch

**Comment by dbort** [comment]
Apart from the existing placeholders, we are not taking any more cherry-picks for this RC. Once those placeholders are resolved and merged, we will make the cut.

**Comment by guangy10** [comment]
We will focus on getting pending CR above merged then do the cut. New cherry-pick beyond this point will not be merged for the final RC. 

If it's critical bug fix, it will need to wait for the batch merge for the final RC before launch.

**Comment by DenisVieriu97** [comment]
Link to landed trunk PR:
- https://github.com/pytorch/executorch/pull/3328

Link to release branch PR:
- https://github.com/pytorch/executorch/pull/3327

Criteria category and reasons:
- Critical bug fix breaking MPS AOT Tracing 

cc @dbort , @guangy10

**Comment by shoumikhin** [comment]
> @shoumikhin Is this a dup of this request [#2666 (comment)](https://github.com/pytorch/executorch/issues/2666#issuecomment-2075614358)?

@DenisVieriu97 if it resolved the CI check https://github.com/pytorch/executorch/actions/runs/8822790798 then yes!

**Comment by DenisVieriu97** [comment]
> @DenisVieriu97 if it resolved the CI check https://github.com/pytorch/executorch/actions/runs/8822790798 then yes!

Thanks @shoumikhin . [test-demo-ios](https://github.com/pytorch/executorch/actions/runs/8822650604/job/24221335959) has passed for that PR run. test-demo-ios will fail in https://github.com/pytorch/executorch/pull/3329 as it was merged before the PR with the fix, but that branch has only MPS doc changes which succeeded in building (cc @guangy10).

**Comment by Olivia-liu** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3229 (manually closed this PR after confirming the changes have benn landed in main)

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3335

Criteria category and reasons:
* Doc fixes

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* N/A

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3338/

Criteria category and reasons:
* Update performance numbers


-----
@mergennachin merged

**Comment by guangy10** [comment]
No outstanding cherry-picks. 
Done creating the "v0.2.0-rc4" tag and kicked off prebuilt binaries from this tag.

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* [placeholder]

Link to release branch PR:
* [placeholder]

Criteria category and reasons:
* Disclaimer

**Comment by dbort** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3345

Link to release branch PR:
* PLACEHOLDER

Criteria category and reasons:
* Fixes tag pattern matching to trigger docs build/push

**Comment by dbort** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3356

Link to release branch PR:
* PLACEHOLDER

Criteria category and reasons:
* Fix flaky macos wheel builds

**Comment by svekars** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3358 
Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3359
Criteria category and reasons:
* Doc build fix

**Comment by dbort** [comment]
Link to landed trunk PR:
* PLACEHOLDER

Link to release branch PR:
* PLACEHOLDER 

Criteria category and reasons:
* Reword the "Preview only" message at the top of the ET website. Instead say that this is an alpha release and that the .pte format and APIs may change in a breaking way before Beta, later in 2024

**Comment by dbort** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3345

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3362
  * NOTE: the previous cherry-pick https://github.com/pytorch/executorch/pull/3347 didn't get all changes in the final trunk PR, so this new PR grabs the rest

Criteria category and reasons:
* Fixes tag pattern matching to trigger docs build/push

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* Pending. @robell 

Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3352

Criteria category and reasons:
* Arm documentation fixes

**Comment by mergennachin** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3374

Link to release branch PR:
* [placeholder]

Criteria category and reasons:
* README documentation

**Comment by dbort** [comment]
No more outstanding cherry-picks. I plan to create the v0.2.0-rc5 tag against 4f798326c6545a58ea958ce3b8fa21f109651747, the current release/0.2 HEAD

**Comment by svekars** [comment]
Link to landed trunk PR:
* https://github.com/pytorch/executorch/pull/3385
Link to release branch PR:
* https://github.com/pytorch/executorch/pull/3387
Criteria category and reasons:
* Doc fix

**Comment by dbort** [comment]
## The [v0.2.0](https://github.com/pytorch/executorch/releases/tag/v0.2.0) release has been cut, and this issue is closed. Use https://github.com/pytorch/executorch/issues/3409 for cherry-picking into the v0.2.1 release.
